runtime/centrifuge/src/migrations.rs

```diff
@@ -203,4 +212,135 @@ mod asset_registry {
 			Ok(())
 		}
 	}
+
+	/// Migrate all the registered asset's metadata to the new version of
+	/// `CustomMetadata` which contains a `AssetTransferability` property. At
+	/// this point in time, the `transferability` of Tranche tokens should be
+	/// set to `AssetTransferability::Xcm` and for all other tokens to
+	/// `AssetTransferability::Xcm`, with the exception of `Currency::Staking`
+	/// tokens which are not registered in the first place.
+	pub struct AssetTransferabilityMigration;
+
+	// The old orml_asset_registry Metadata storage using v0::CustomMetadata
+	#[storage_alias]
+	type Metadata<T: orml_asset_registry::Config> = StorageMap<
+		orml_asset_registry::Pallet<T>,
+		Twox64Concat,
+		CurrencyId,
+		AssetMetadata<Balance, v0::CustomMetadata>,
+		OptionQuery,
+	>;
+
+	impl OnRuntimeUpgrade for AssetTransferabilityMigration {
+		fn on_runtime_upgrade() -> Weight {
+			orml_asset_registry::Metadata::<Runtime>::translate(
```

Comment:
It would be great if we made this migration conditional, e.g. on some pallet version. If for some reason this migration is not removed before the next RU, it panics as it fails to decode the values of the storage map.

However, we cannot use the `orml_asset_registry` version as this must not be overwritten by us. Otherwise, we will have harder times running default (from ORML itself) `orml_asset_registry` migrations in the future.

We can take a look at this later as I don't have a solution at the top of my head. We should think about whether we want to introduce versioning to our custom metadata.

Reply:
Thanks for the thoughts here. Can't we just use the runtime version the migration is running on to ensure 1) that it's being executed against the correct version and 2) that it won't be running again even if not deleted for a follow up RTU?

Reply:
Great idea. I think that totally suffices!